### ISSUE TITLE
log_rotationdir is not recognized as a valid parameter

### ISSUE THREAD

**Issue by rrod157** [biemond/biemond-orawls#528]
I am using v2.0.15. When i added it in on wls_server command, it seem to fail with the following error:

> Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: no parameter named 'log_rotationdir'

Apparently, I am fixing an issue introduced by this attribute since we upgraded orawls and now it is not recognizable. I checked that log_rotationdir.rb is defaulting it to blank (`defaultto ''`) but it cannot accept custom value:

`log_rotationdir  => "/appl/spool/logs/${domain_name}/servers/AdminServer",`

 How this can be handled?

**Comment by biemond** [comment]
are you using a puppet master, if so it is a caching issue. please check the readme of orawls

**Comment by userMar2017** [comment]
Ed,

To provided more details, we were on 2.0.6 for months and our manifests worked just fine. Since 2.0.6 does not support 12.2.1.x, on your advise we upgraded to 2.0.15. After upgrading, we realized that it started throwing the error listed below

.. puppet-agent[XXXXX]:<admin_server>/log_rotationdir) log_rotationdir changed '/some_folder/somefile' to ''

We then proceeded to add option "log_rotationdir" to "wls_server" but it did not accept it and threw the following exception

...puppet-agent[XXXX]: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: no parameter named 'log_rotationdir' (file:...

**Comment by biemond** [comment]
yep follow the puppet master tips, it is a cache issue.  puppet apply will work fine.

**Comment by userMar2017** [comment]
If you still think its related to caching, let us know if you want us to restart all the PE services on the puppet master. Problem is that we have a number of branches / environments ; do you think we will need to perform this on all the environments ? We did not encounter this issue in the branch we tested this on.

After adding or refreshing the easy_type or orawls modules you need to restart all the PE services on the puppet master (this will flush the PE cache) and always do a puppet agent run on the Puppet master

**Comment by biemond** [comment]
do some dummy puppet agent runs on puppet master so it detect the new modules, put it on the primary location. 
after that it will sync the agents with the new module versions

**Comment by userMar2017** [comment]
Thanks for the quick responses Ed. However, before I go to our Puppet team, I will need to justify to them why we need to perform these steps on the master.

1. Its odd that everything else works except that we run into this issue. If I do a "puppet agent -t", will it help confirm that this is a caching issue ?
2. If its a caching issue, are you asking us to run a "puppet agent -t" on the master next ?

**Comment by biemond** [comment]
somehow the puppet types changes are not detected , manifests are fine. 
just do some puppet agent  run on the master with some user or group or maybe empty will already work.

**Comment by userMar2017** [comment]
So, I should ask them to run "puppet agent -t" on the master against this particular branch ?

**Comment by userMar2017** [comment]
Puppet enginner restarted pe-puppetserver but I encountered the same issue. Opened a ticket with Puppet.

**Comment by biemond** [comment]
ok but do you have multiple versions of orawls on the  same master

**Comment by userMar2017** [comment]
I will pass on your feedback Ed. Thanks a lot for all of your support. Really appreciate it.

**Comment by bFekete** [comment]
@userMar2017 what version of Puppet are you using? I believe this was fixed in PE 2016.5 / Puppet 4.8.

**Comment by userMar2017** [comment]
We are on PE 2017.3 actually

**Comment by userMar2017** [comment]
Ed, with this update, is "log_rotationdir" a mandatory argument ? Do I need to add it to all the admin + managed servers ?

**Comment by userMar2017** [comment]
If I dont modify it, it causes error " log_rotationdir changed '_actual_location_' to '"
This in turn causes modification to config.xml ; the actual change is "The non-dynamic attribute CustomIdentityKeyStorePassPhraseEncrypted on weblogic.management.configuration.ServerMBeanImpl(blah blah) has been changed. "
This change caused a number of issues for us in production as we had a number of developers complain that their deployment task failed.
I will be adding this attribute and attempting to upgrade orawls a second time later this week.

**Comment by biemond** [comment]
or remove this line https://github.com/biemond/biemond-orawls/blob/puppet4/lib/puppet/type/wls_server/log_rotationdir.rb#L6

**Comment by userMar2017** [comment]
Is there a new orawls module for this ? I hate to modify it manually on our servers.

**Comment by userMar2017** [comment]
I dont even have access to the Puppet master and the puppet admin is not too happy with me !

**Comment by biemond** [comment]
first check if this solves your issue

**Comment by userMar2017** [comment]
I cannot reproduce this error using puppet apply ; let me ask the puppet admin for this change

**Comment by userMar2017** [comment]
Yes, that did the trick. I had to get it modified on the puppet master and after that it worked. It threw the following message which suggests it implemented the change. The original error "log_rotationdir changed" is now gone !

   desc 'The log rotation dir of the server'
 
-  defaultto ''
-
   to_translate_to_resource do | raw_resource|
     raw_resource['log_rotationdir']
   end

**Comment by biemond** [comment]
good I shall remove it, normally no one wants the default log location and that is the reason why we use wls_servers

**Comment by userMar2017** [comment]
Thanks Ed. Let me know which version of the module to use.

**Comment by rrod157** [comment]
Hi Ed, would like to know when the new version would be available? Thanks for all your help.

**Comment by biemond** [comment]
Done, it should be there

**Comment by userMar2017** [comment]
I will go with 2.0.16. thanks ed !
